cloud/docker/docker_image.py

```diff
@@ -371,12 +371,6 @@ def push_image(self, name, tag=None):
         registry, repo_name = auth.resolve_repository_name(repository)
 
         if re.search('/', repository):
-            if registry:
```

Comment:
Thanks for your feedback. Would you mind providing more information here, or a code snippet? Unfortunately I don't understand what is mean by, "keep a flag and then return it if/when it fails to connect." I am extremely new to Ansible's codebase, so I don't know yet the idea of returning values and what that does to the flow.

Would returning such a value cause the task to be marked as a failure? I want to avoid the task being marked as a failure in cases where no credentials were needed in the first place.

My thinking here is that an exception raised by docker-py should handle the case of terminating the play when no credentials were found but they were indeed necessary. I would be amenable to keeping this logic and downgrading `self.fail` to `self.warn` (if there is such a method?), however, I refer back to this portion of the PR's original description (emphasis added):

> the docker-py client will need to do its own management of finding credentials, anyway—an implementation detail that is subject to change and could easily diverge from the check in docker_image.

Thus, docker_image could end up issuing warnings when they are unwarranted because a different scheme is used to collect credentials by the docker-py client.